ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -1296,8 +1312,24 @@ func (h *FaultHandler) startNetworkPacketLossFault(ctx context.Context, taskMeta
 		field.CommandOutput: string(cmdOutput[:]),
 	})
 	// After creating the queueing discipline, create filters to associate the IPs in the request with the handle.
+	// First redirect the allowlisted ip addresses to band 1:3 where is no network impairments.
+	for _, ip := range request.SourcesToFilter {
+		tcAllowlistIPCommandComposed := nsenterPrefix + fmt.Sprintf(tcAllowlistIPCommandString, interfaceName, aws.StringValue(ip))
+		cmdList = strings.Split(tcAllowlistIPCommandComposed, " ")
+		_, err = h.runExecCommand(ctx, cmdList)
+		if err != nil {
+			logger.Error("Command execution failed", logger.Fields{
+				field.CommandString: tcAllowlistIPCommandComposed,
+				field.Error:         err,
+				field.CommandOutput: string(cmdOutput[:]),
+				field.TaskARN:       taskMetadata.TaskARN,
+			})
+			return err
+		}
+	}
+	// After processing the allowlisted ips, associate the ip addresses in Sources with the qdisc.
```

Comment:
Is this identical to the code block added to `startNetworkLatencyFault` method? If so, maybe we should create a helper function? These methods are already quite heavy.

Reply:
Thanks for the callout. Will add this to a helper function